### ISSUE TITLE
x86_64 macOS Unable to Use Latest PyTorch via DJL in QuPath 0.6

### ISSUE THREAD

**Issue by jaredbrewer** [qupath/qupath#1733]
Before submitting your bug report, please check the following:

* [x] I've definitely found a bug (it you're not sure, please use [image.sc](https://forum.image.sc/tags/qupath) instead)
* [x] I've checked https://qupath.github.io for a new release that might already have fixed the issue
* [x] I've checked the [Changelog](https://github.com/qupath/qupath/blob/master/CHANGELOG.md) to see if the bug has already been fixed in the next release
* [x] I've checked for existing GitHub issues describing the same problem

### Describe the bug
QuPath very nicely continues to support x86-based macOS systems but the implementation of the DeepJavaLibrary in tandem with PyTorch's deprecation of x86-based macOS support means that PyTorch does not work within QuPath and, thus, InstanSeg.

### To Reproduce
On a macOS x86_64-based system, attempt to download PyTorch from the Extension > DeepJavaLibrary > Manage DJL engines dialog. This will fail without a meaningful error beyond "Unable to initialize PyTorch, sorry."

### Expected behavior
There are a few potential options here:
- Automatically download the latest compatible version of PyTorch for a given operating system and architecture
- Provide additional instruction about how to download and use a different version of PyTorch within QuPath
- Create a TensorFlow-compatible version of InstanSeg, which remains compatible with Intel Macs and there are no plans (to my knowledge) to deprecate this support in the immediate future.

### Desktop (please complete the following information):
 - OS: macOS 15.2 Sequoia
 - QuPath Version: 0.6.0-rc3

### Additional context
While Apple no longer sells new Macs running on Intel, there remains a large install base and specific use cases where an ARM-based Mac is not a viable alternative. PyTorch's choice to deprecate these systems in early 2024 was disappointing but most of the major features and models remain freely compatible among PyTorch 2.x version (for now). It would be useful to provide as much support as possible for the systems that remain in use.

**Comment by petebankhead** [comment]
I've removed the 'bug' label since it was a conscious decision, but I've made a PR to address this at https://github.com/qupath/qupath/pull/1734

The PR aims to prolong Intel support a bit longer, but it's a delicate balance. If we go back too far with the DJL version, then TensorFlow doesn't work at all on Apple Silicon. In fact there is only *one* version that supports both PyTorch on Intel and TensorFlow on Apple Silicon (v0.29.0), but even that requires a system property to be set. Since there are continually improvements and fixed, we don't want to be stuck on that version indefinitely.

The PR tries to compromise by using the latest DJL but requesting that an older PyTorch is used *only* for Intel Mac.

It works for me on my M3 Mac - both building for Apple Silicon and Intel (via Rosetta2). But I can't test it on my 2014 MacBook Pro because it's stuck on macOS 11, and it seems DJL/PyTorch gave up on it much further in the past... and I don't see any way around that.

**Comment by alanocallaghan** [comment]
> Provide additional instruction about how to download and use a different version of PyTorch within QuPath

For what it's worth, using [QuBaLab](https://qupath.github.io/qubalab-docs/) it's possible to use any version of PyTorch with QuPath, albeit not with the same UI integration as in the InstanSeg extension. You can find some example code in the i2k talk materials here: https://github.com/qupath/i2k-qupath-for-python-programmers/blob/main/clustering-objects.ipynb

or follow-along instructions from virtual i2k: https://www.youtube.com/watch?v=984Jz2QCvng

This will be substantially slower than other means of running InstanSeg, but does mean that you *can* run instanseg even on any system that supports PyTorch >= 2 (as this is required for InstanSeg generally) even if completely abandoned by DJL et al.

**Comment by petebankhead** [comment]
I'll close this, because I've merged the PR and it should be possible to use InstanSeg with Intel Macs as long as they are running macOS 12+.

We'll try to maintain support for as long as it's possible without restricting our ability to include more recent PyTorch/DJL fixes or add new features that work on all the other platforms. I think the PR achieves that currently, although I don't have a suitable Intel Mac to test it on.

**Comment by jaredbrewer** [comment]
These all seem like reasonable solutions to me, I appreciate your efforts to look into these issues and provide a solution.
